caps/capabilities.go

```diff
@@ -222,3 +236,17 @@ func (e *NotFoundError) Error() string {
 		panic(fmt.Sprintf("unexpected what: %q", e.what))
 	}
 }
+
+// Validate if a capability is correct according to the given type
+func (t Type) Validate(c *Capability) error {
+	if t != c.Type {
+		return fmt.Errorf("capability is not of type %q", t)
+	}
+	// While we don't have any support for type-specific attribute schema,
```

Comment:
Right now my idea is that it will be a JSON schema. Perhaps we'll add a few more format checkers that are domain specific "e.g. path to a character device" or "path to a socket" that would make it better at describing the problem space.

In the meantime (for this sprint) I'll probably just test that only certain attributes are set and we'll work with that for as long as external definition for types is born.